js/world.js

```diff
@@ -0,0 +1,33 @@
+class World {
+    constructor(graph, roadWidth = 100, roadRoundness = 10)  {
+        this.graph = graph;
+        this.roadWidth = roadWidth;
+        this.roadRoundness = roadRoundness;
+
+        this.envelopes = [];
+
+        this.generate();
+    }
+    //for each segments in the graph, make a envelope around them, then store them in the envelopes array
+    generate()  {
+        //because we are re-generating the envelope 60 times/frame, empty out the envelopes to begin with. 
+        this.envelopes.length = 0;
+        for (const seg of this.graph.segments) {
+            this.envelopes.push(
+                new Envelope(seg, this.roadWidth, this.roadRoundness)
+            );
+        }
+        //attribute to figure out the intersections of the two segments and break them. 
+        this.intersections = Polygon.break(
```

Comment:
as of right now, it only works for the first and the second envelopes. 
To do : apply it to all of the envelopes. 
![Screenshot 2024-03-08 at 5 34 00 PM](https://github.com/lucyJihyeon/Self-Driving/assets/141371734/1900a349-60ce-44b2-a428-010419a3f95a)
in the screen shoot, we can see the color changes between the very left envelope and the bottom envelope, which means the segment successfully broken into, but not with the right one.